src/components/StudyPane/InfoPane/Motif/RootBlock.tsx

```diff
@@ -66,36 +76,99 @@ export const RootBlock = ({
     setSelected(hasChildren);
   }, [ctxSelectedHebWords, descendants]);
 
-  const handleClick = (e: React.MouseEvent) => {
-    setSelected(prevState => !prevState);
-
-    let updatedSelectedHebWords = [...ctxSelectedHebWords];
-    if (!selected) {
-      updatedSelectedHebWords = ctxSelectedHebWords.concat(descendants);
-    } else {
-      descendants.forEach((dsd) => {
-        updatedSelectedHebWords.splice(updatedSelectedHebWords.indexOf(dsd), 1)
-      })
-    }
-    ctxSetSelectedHebWords(updatedSelectedHebWords);
-    ctxSetNumSelectedWords(updatedSelectedHebWords.length);
-  };
-
   useEffect(() => {
-    if (ctxColorAction !== ColorActionType.none && selected) {
-      if (ctxColorAction === ColorActionType.colorFill && ctxSelectedColor) {
-        setColorFillLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.borderColor && ctxSelectedColor) {
-        setBorderColorLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.textColor && ctxSelectedColor) {
-        setTextColorLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.resetColor) {
-        setColorFillLocal(DEFAULT_COLOR_FILL);
-        setBorderColorLocal(DEFAULT_BORDER_COLOR);
-        setTextColorLocal(DEFAULT_TEXT_COLOR);
+    if (selected){
+      const matchedFillColor = matchFillColor();
+      const matchedTextColor = matchTextColor();
+      const matchedBorderColor = matchBorderColor();
+      const rootBlockColor = ctxRootsColorMap.get(descendants[0].strongNumber);
+      let colorObject: ColorType = {} as ColorType;
+      if (ctxColorAction !== ColorActionType.none && selected) {
+        if (ctxColorAction === ColorActionType.colorFill && ctxSelectedColor) {
```

Comment:
See if you could refactor this block of code. I'm seeing many repetition of updates and loops.